dds/DCPS/Qos_Helper.inl

```diff
@@ -927,22 +927,87 @@ bool Qos_Helper::valid(const DDS::DomainParticipantQos& qos)
 ACE_INLINE
 bool Qos_Helper::valid(const DDS::TopicQos& qos)
 {
-  return
-    valid(qos.topic_data)
-    && valid(qos.durability)
+  if (!valid(qos.topic_data))
+    ACE_ERROR_RETURN ((LM_ERROR,
+        ACE_TEXT("(%P|%t) ERROR: ")
+        ACE_TEXT("Qos_Helper::valid::TopicQos, ")
+        ACE_TEXT("invalid topic_data qos.\n")), false);
```

Comment:
```suggestion
  if (!valid(qos.topic_data)) {
    if (log_level >= LogLevel::Error) {
      ACE_ERROR((LM_ERROR,
        "(%P|%t) ERROR: Qos_Helper::valid: invalid topic_data qos.\n"));
    }
    return false;
  }
```

ACE_TEXT isn't needed.
ACE_ERROR_RETURN is something we are trying to avoid using because it hides return statements behind functions.
The log_level code helps with avoiding debug message spam.  You will need to add #include "debug.h" for this code.
And then a few formatting changes.

@iguessthislldo is invalid QoS a Notice or an Error?

Reply:
Should I apply these changes to the whole file or just what I originally changed in the PR (and by that I mean the other logging that was added a while ago)

Reply:
It should probably be a notice because it's reported back to the user through the error status and it's not fatal to the program or causes OpenDDS to function incorrectly, though the part about misconfiguration in the guidelines makes this a little muddled. [These are the guidelines we are referencing btw](https://opendds.readthedocs.io/en/master/internal/dev_guidelines.html#logging).

You don't have to apply these changes to the whole file if you don't want to. I hope to automate it someday and do it all at once, but in the mean time it's been something we've been trying to do incrementally for the whole code base for the most part.

Reply:
I'll change them to the be notices, and I've already started to change the whole file as it wasn't a huge deal. Also as for the formatting issues, is this the script to auto format it correctly, https://github.com/objectcomputing/OpenDDS/blob/master/tools/scripts/style/style.sh.

Reply:
Okay, thanks. That script it hasn't been maintained in a while, so I can't really recommend using it as I'm not sure it follows our current practices. [`tools/scripts/lint.pl`](https://github.com/objectcomputing/OpenDDS/blob/master/tools/scripts/lint.pl) is maintained and runs as part of the PR checks (in the "lint" job). It can fix things like trailing spaces on lines if `--try-fix` is passed, but it's mostly just meant to report issues.